strawberryfields/backends/tfbackend/states.py

```diff
@@ -55,124 +44,72 @@ def __init__(
-
-            if evaluate:
-                t = session.run(t, feed_dict=feed_dict)
-                if close_session:
-                    session.close()
-            return t
 
     def trace(self, **kwargs):
         r"""
         Computes the trace of the state. May be numerical or symbolic.
 
         Args:
-            **kwargs: Optional keyword arguments.
-
-                * If this contains the key
-                  ``eval``, then the corresponding argument will be used to determine the return behaviour of this function.
-                  When ``eval=True``, the return value is numerical; when ``eval=False``, it is symbolic.
-                * If eval is not present in kwargs, then state falls back to the an internal evaluation behaviour,
-                  which is specified at initialization.
-                * A Tensorflow Session or feed_dict may also be passed via the keys ``session`` or ``feed_dict``, respectively.
-                  If a Session is supplied, then ``eval`` is overriden and the numerical evaluation takes place in the provided Session.
-                  If session and/or feed_dict are not given, then a temporary session and/or empty feed_dict will be used.
 
         Returns:
             float/Tensor: the numerical value, or an unevaluated Tensor object, for the trace.
         """
-        with self._graph.as_default():
-            s = self.data
-            if not self.batched:
-                s = tf.expand_dims(s, 0)  # insert fake batch dimension
-            if self.is_pure:
-                flat_state = tf.reshape(s, [-1, self.cutoff_dim ** self.num_modes])
-                tr = tf.reduce_sum(flat_state * tf.conj(flat_state), 1)
-            else:
-                # hack because tensorflow einsum doesn't allow partial trace
-                tr = s
-                for _ in range(self.num_modes):
-                    tr = tf.trace(tr)
-            if not self.batched:
-                tr = tf.squeeze(tr, 0)  # drop fake batch dimension
-
-            tr = self._run(tr, **kwargs)
-            return tr
+        s = self.data
+        if not self.batched:
+            s = tf.expand_dims(s, 0)  # insert fake batch dimension
+        if self.is_pure:
+            flat_state = tf.reshape(s, [-1, self.cutoff_dim ** self.num_modes])
+            tr = tf.reduce_sum(flat_state * tf.math.conj(flat_state), 1)
+        else:
+            # hack because tensorflow einsum doesn't allow partial trace
+            tr = s
+            for _ in range(self.num_modes):
+                tr = tf.linalg.trace(tr)
+        if not self.batched:
+            tr = tf.squeeze(tr, 0)  # drop fake batch dimension
+
+        return tr
 
     def fock_prob(self, n, **kwargs):
         r"""
         Compute the probabilities of a specific Fock-basis matrix element for the state. May be numerical or symbolic.
 
         Args:
             n (Sequence[int]): the Fock state :math:`\ket{\vec{n}}` that we want to measure the probability of
```

Comment:
```suggestion
            n (Sequence[int]): the Fock state :math:`\ket{\vec{n}}` that we want to measure the probability of.
```

Reply:
Since this is a sentence fragment, I think it's fine to not have the period